src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -1675,8 +1676,18 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       //
       //   // Later generated code expects foo.bar = {} to exist:
       //   foo.bar.Baz = function() { /* ... */ }
-      printer->Print("goog.exportSymbol('$name$', null, global);\n", "name",
-                     *it);
+
+      // Do not use global scope in strict mode
+      if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
+        string namespaceObject = *it;
+        // Remove "proto." from the namespace object
```

Comment:
We should verify (GOOGLE_CHECK()?) that the string we're removing is actually "proto.".

Reply:
Added

Reply:
I don't see anything added? Once this is added, I approve the change to go in.

Reply:
@haberman how about now?